tests/sphinxext/test_workchain.py

```diff
@@ -8,31 +8,27 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """Tests for the AiiDA workchain Sphinx directive."""
+import pytest
 
-from os.path import join, dirname
 
-WORKCHAIN = join(dirname(__file__), 'workchain_source')
-
-
-def test_workchain_build(build_sphinx, xml_equal, reference_result):
+def test_workchain_build(sphinx_build_factory, xml_equal, reference_result):
     """Test building sphinx documentation for WorkChain.
 
     Builds Sphinx documentation for workchain and compares against expected XML result.
     """
-    out_dir, completed_proc = build_sphinx(WORKCHAIN)
-    assert completed_proc.returncode == 0
-    index_file = join(out_dir, 'index.xml')
+    sphinx_build = sphinx_build_factory('workchain', buildername='xml')
```

Comment:
@greschd you just change the `buildername` here

Reply:
Hmm, I think the reason I had `xml` here initially is because the HTML output would be too flimsy a test (slight changes and such). If your (more recent) experience is that it's not it would be fine to change. 
Otherwise I'd prefer keeping a way to build the "test documentations" outside of the actual tests - the reason being that is that it's a nice way to iterate on the design of the HTML.

Reply:
Which is why I mentioned using beautifulsoup; to sanitise the output before comparison.

If you want to build the docs, just run: `sphinx-build -b html -n -W --keep-going tests/sphinxext/sources/workchain tests/sphinxext/sources/workchain/build`, you don't need a Makefile.

Note at present, you would have to add to conf.py something like:

```python
import sys
sys.path.append("/Users/chrisjsewell/Documents/GitHub/aiida_core_develop/tests/sphinxext/workchains")
from aiida.manage.configuration import load_documentation_profile
load_documentation_profile()
```

The latter part we should actually add to the extension (as an optional step governed by a config value), because I imagine you can't actually use it without running that.

alternatively to iterate on the design (this is how I do it), just paste somewhere in the docs, e.g.

```
.. aiida-calcjob:: CalcJob
    :module: aiida.engine.processes.calcjobs
    :expand-namespaces:
```

and run `tox py38-docs-live`, and you get live updates of the docs builds in the browser

Reply:
> The latter part we should actually add to the extension (as an optional step governed by a config value), because I imagine you can't actually use it without running that.

Actually, you can. The extension itself doesn't rely on the profile being loaded, so if the `.spec()` of the plugin can be created without it it'll work. But yeah, I think 99% of plugins will use it so we might as well add that.

> and run tox py38-docs-live, and you get live updates of the docs builds in the browser

Good enough for me :+1: